docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+These are the exact reasons why Nimbus itself is in the application-services repo.
+
+### Option 3: Use the existing nimbus client
+Splitting the [existing client](https://github.com/mozilla/application-services/blob/6069bd26d742f9597ab7e62d8df23ad634c7bd98/components/nimbus/src/client/http_client.rs#L9) out from Nimbus in a way that allows Nimbus to continue to use it, while also making it available for stand-alone use is also an attractive option.
+
+In particular, the feature set of that client overlaps with the requirements of the new library - no local persistence is necessary and no signature verification is required. It already works with Viaduct and is used by a component which is exposed via UniFFI.
+
+## Chosen Option
+
+We have chosen Option 3 because it allows us to reuse the new client in Nimbus, as well as on iOS and on Android with minimal initial development effort.
+If the new library ends up growing requirements that are already in the existing remote settings client, we remain able to copy that functionality from that library into this.
+
+## Specific Plans
+
+This section is non-normative - ie, is not strictly part of the ADR, but exists
+for context.
+
+This is a very high-level view of the tasks required here.
+
+* Create a new top-level component in the application-services repository (ie, under application-services/components), identify the exact API we wish to expose for this new library, describe this API using UniFFI, then implement the API with “stubs” (eg, having each method just be a rust todo!() macro or similar.)
+* Identify which parts of Nimbus should be factored out into a shared component (which would live under application-services/components/shared) and move that functionality to the new shared component. This component probably will not have a UniFFI .udl file, but is just for consumption by the new component above and the existing nimbus component (although this is up in the air slightly - if it is a requirement that nimbus and the new component share some configuration or initialization code, we might need to do something more complex here. This seems unlikely, but possible, so is included here for completeness). It is expected that a number of tests will also move here.
+* Update Nimbus to take a dependency on the new component and use it, including tests.
+* Flesh out the API of the new top-level component using the new shared component (ie, replace the todo!() macros with real code.)
+* Identify any changes necessary to the Nimbus android/swift code necessary for this - in particular, any shared configuration and initialization code identified above in the application-services repo.
+* Implement the Android and iOS code in the application-services repo desired to make this an ergonomic library for the mobile platforms.
+* Update the mobile code for the UniFFI changes make to Nimbus, if any.
+* Implement the mobile code which consumes the new library, including tests.
+* Profit?
```

Comment:
(not sure where to put a more general question, so putting it here) Are there concerns or mitigations for when the library has to pull a large set of changes -- like the initial pull for data for Suggest offline users? There were concerns about shipping this amount of data with the applications as well as downloading it on first run, and I'm curious if this addresses that or if that's up to the clients to solve.

Reply:
That's a great question - IIUC, Nimbus has the same general requirement too. I suspect that addressing this will force us to rethink the "doesn't need local persistence" requirement in the future. I'd welcome any suggested changes I should make here.